language/move-binary-format/src/file_format.rs

```diff
@@ -251,6 +251,28 @@ pub struct StructHandle {
     pub abilities: AbilitySet,
     /// The type formals (identified by their index into the vec) and their constraints
     pub type_parameters: Vec<AbilitySet>,
+    /// The kind of each formal (either phantom or non-phantom)
+    pub type_param_kinds: Vec<TypeParamKind>,
+}
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+pub enum TypeParamKind {
```

Comment:
oof, `kind` is a bit overloaded in the history of this project so this might be a bit confusing. 

I would suggest
```
TypeParameterUsage {
   Value, 
   Phantom,
}
```